AssetsPickerViewController/Classes/Photo/Controller/AssetsPhotoViewController.swift

```diff
@@ -712,21 +712,40 @@ extension AssetsPhotoViewController: AssetsAlbumViewControllerDelegate {
 // MARK: - AssetsManagerDelegate
 extension AssetsPhotoViewController: AssetsManagerDelegate {
     
+    private func shouldUpdateAssetCollectionView(oldStatus: PHAuthorizationStatus, newStatus: PHAuthorizationStatus) -> Bool {
+        if #available(iOS 14, *) {
+            if oldStatus != .limited || oldStatus != .authorized {
+                if newStatus == .limited || newStatus == .authorized {
+                    return true
+                }
+            }
+        } else {
+            if oldStatus != .authorized {
+                if newStatus == .authorized {
+                    return true
+                }
+            }
+        }
+        return false
+    }
+    
     public func assetsManager(manager: AssetsManager, authorizationStatusChanged oldStatus: PHAuthorizationStatus, newStatus: PHAuthorizationStatus) {
-        if oldStatus != .authorized {
-            if newStatus == .authorized {
+        if shouldUpdateAssetCollectionView(oldStatus: oldStatus, newStatus: newStatus) {
                 updateNoPermissionView()
                 AssetsManager.shared.fetchAssets(isRefetch: true, completion: { [weak self] (_) in
                     self?.collectionView.reloadData()
                 })
-            }
+            
         } else {
             updateNoPermissionView()
         }
     }
     
-    public func assetsManager(manager: AssetsManager, reloadedAlbumsInSection section: Int) {}
-    public func assetsManager(manager: AssetsManager, insertedAlbums albums: [PHAssetCollection], at indexPaths: [IndexPath]) {}
+    public func assetsManager(manager: AssetsManager, reloadedAlbumsInSection section: Int) {
+    }
+    public func assetsManager(manager: AssetsManager, insertedAlbums albums: [PHAssetCollection], at indexPaths: [IndexPath]) {
+        
+    }
```

Comment:
These delegate methods are all empty. Do we still need these changes?

Reply:
I've removed the extra newline